parsons/shopify/shopify.py

```diff
@@ -0,0 +1,192 @@
+        self.api_version = check_env.check('SHOPIFY_API_VERSION', api_version)
+
+    def get_count(self, query_date=None, since_id=None, table_name=None):
+        """
+        Get the count of rows in a table.
+
+        `Args:`
+            query_date: str
+                Filter query by a date that rows were created. This filter is ignored if value
+                is None.
+            since_id: str
+                Filter query by a minimum ID. This filter is ignored if value is None.
+            table_name: str
+                The name of the Shopify table to query.
+        `Returns:`
+            int
+        """
+        return requests.get(self.get_query_url(query_date, since_id, table_name), auth=(self.api_key, self.password)).json().get("count", 0)
+
+    def get_orders(self, query_date=None, since_id=None, completed=True):
+        """
+        Get Shopify orders.
+
+        `Args:`
+            query_date: str
+                Filter query by a date that rows were created. Format: yyyy-mm-dd. This filter
+                is ignored if value is None.
+            since_id: str
+                Filter query by a minimum ID. This filter is ignored if value is None.
+            completed: bool
+                True if only getting completed orders, False otherwise.
+        `Returns:`
+            Table Class
+        """
+        orders = []
+
+        def _append_orders(url):
+            nonlocal orders
+
+            if completed:
+                url += '&financial_status=paid'
+
+            res = requests.get(url, auth=(self.api_key, self.password))
+
+            cur_orders = res.json().get("orders", [])
+
+            # Flatten orders to non-complex types
+            for order in cur_orders:
+                keys_to_add = {}
+                keys_to_delete = []
+
+                for key1 in order:
+                    if isinstance(order[key1], dict):
+                        for key2 in order[key1]:
+                            keys_to_add[key1 + '_' + key2] = order[key1][key2]
+                        keys_to_delete.append(key1)
+                    elif key1 == 'note_attributes':
+                        for note in order[key1]:
+                            keys_to_add[key1 + '_' + note['name']] = note['value']
+
+                order.update(keys_to_add)
+                for key in keys_to_delete:
+                    del order[key]
+
+            orders += cur_orders
+
+            return res
+
+        res = _append_orders(self.get_query_url(query_date, since_id, "orders", False))
+
+        # Get next page
+        while res.headers.get("Link"):
+            link = re.split('; |, ', res.headers.get("Link"))
+            if len(link) and link[len(link) - 1] == 'rel="next"':
+                res = _append_orders(link[len(link) - 2][1:-1])
+            else:
+                break
+
+        tbl = Table(orders)
+
+        # Fill null vals
+        for col in tbl.columns:
```

Comment:
I don't have a strong preference, but why wouldn't you just leave the values as `None` here?

Reply:
Good call!